src/operators.jl

```diff
@@ -235,8 +236,8 @@ Flatten a signal of signals into a signal which holds the
 value of the current signal. The `typ` keyword argument specifies
 the type of the flattened signal. It is `Any` by default.
 """
-function flatten(input::Signal; typ=Any)
-    n = Signal(typ, value(value(input)), (input,))
+function flatten(input::Signal; typ=Any, name=auto_name())
+    n = Signal(typ, value(value(input)), (input,); name=name)
```

Comment:
I think naming them on creation makes sense, and the kw args are already there now. I've also added a `rename!(s::Signal, name::String)`